build-aux/main.mk

```diff
@@ -95,12 +94,19 @@ build: pkg/install/helm/telepresence-chart.tgz ## (Build) Build all the source c
 	sed $(foreach v,TELEPRESENCE_REGISTRY TELEPRESENCE_BASE_VERSION, -e 's|@$v@|$($v)|g') <$< >$@
 .PHONY: image push-image
 image: .ko.yaml $(tools/ko) ## (Build) Build/tag the manager/agent container image
-	localname=$$(GOFLAGS="-ldflags=-X=$(PKG_VERSION).Version=$(TELEPRESENCE_VERSION) -trimpath" GOOS=linux ko publish --local ./cmd/traffic) && \
-	docker tag "$$localname" $(TELEPRESENCE_REGISTRY)/tel2:$(patsubst v%,%,$(TELEPRESENCE_VERSION))
+	localname=$$(GOFLAGS="-ldflags=-X=$(PKG_VERSION).Version=$(TELEPRESENCE_VERSION) -trimpath" ko publish --platform=linux/amd64 --local ./cmd/traffic) && \
+	docker tag "$$localname" $(TELEPRESENCE_REGISTRY)/tel2:$(patsubst v%,%,$(TELEPRESENCE_VERSION))-amd64
+	localname=$$(GOFLAGS="-ldflags=-X=$(PKG_VERSION).Version=$(TELEPRESENCE_VERSION) -trimpath" ko publish --platform=linux/arm64 --local ./cmd/traffic) && \
```

Comment:
Very odd, but apparently, this makes `ko publish` build using `linux/amd64` in spite of the platform argument. The output from CI shows:
```console
localname=$(GOFLAGS="-ldflags=-X=github.com/telepresenceio/telepresence/v2/pkg/version.Version=v2.0.0-gotest.8b373a00308a4a257b5baa686b23abce8490a8ee -trimpath" ko publish --platform=linux/arm64 --local ./cmd/traffic) && \
docker tag "$localname" docker.io/datawire/tel2:2.0.0-gotest.8b373a00308a4a257b5baa686b23abce8490a8ee-arm64
2021/09/11 20:22:41 Using base docker.io/datawire/tel2-base:0d118a970c93bb1387fce6bb5638e86eb82d2cbe for github.com/telepresenceio/telepresence/v2/cmd/traffic
2021/09/11 20:22:42 Building github.com/telepresenceio/telepresence/v2/cmd/traffic for linux/amd64
```
and errors with:
```consoe
Error response from daemon: reference does not exist
```
Perhaps GOHOSTARCH must be explicitly set?

Reply:
I can't reproduce on my dev environment. Do you know what platform/environment/distro the CI is running?

Reply:
It's using go version go1.17 linux/amd64 on Ubuntu 20.04.